src/providers/ldap/sdap_async.c

```diff
@@ -133,14 +134,41 @@ static void sdap_ldap_next_result(struct tevent_context *ev,
     sdap_process_result(ev, pvt);
 }
 
+static struct sdap_op *sdap_get_message_op(struct sdap_handle *sh,
+                                           LDAPMessage *msg)
+{
+    struct sdap_op *op;
+    int msgid;
+
+    msgid = ldap_msgid(msg);
+    if (msgid == -1) {
+        DEBUG(SSSDBG_OP_FAILURE, "can't fire callback, message id invalid!\n");
```

Comment:
IMO, it's better to update message text.

Reply:
Thank you. Fixed.